controllers/actions.github.com/clientutil.go

```diff
@@ -20,3 +20,13 @@ func patch[T object[T]](ctx context.Context, client patcher, obj T, update func(
 	update(obj)
 	return client.Patch(ctx, obj, kclient.MergeFrom(original))
 }
+
+type subResourcePatcher interface {
```

Comment:
do we still need the other `patcher` in this file?

Reply:
Apparently not- let me remove it! Thanks for pointing it out ☺️

Reply:
Oops sorry! I realized that you still use it in various controllers for autoscalinglistener, autoscalingrunnerset, ephemeralrunner and ephemeralrunnersets. We'd better leave the `patcher` interface along with the `patch` function that uses the interface for those controllers.

Reply:
Sorry, my bad, ignore this.